lib/datadog/open_feature/exposures/reporter.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require_relative 'event'
+require_relative 'deduplicator'
+
+module Datadog
+  module OpenFeature
+    module Exposures
+      # This class is responsible for reporting exposures to the Agent
+      class Reporter
+        def initialize(worker, telemetry:, logger:)
+          @worker = worker
+          @logger = logger
+          @telemetry = telemetry
+          @deduplicator = Deduplicator.new
+        end
+
+        def report(result, flag_key:, context:)
+          return false unless result.log?
+          return false if context.nil?
+
+          event = Event.build(result, flag_key: flag_key, context: context)
+          return false if @deduplicator.duplicate?(event)
```

Comment:
Most events will be built just to be discarded. I see in `Event.build` that it creates quite a few Ruby objects.

Do you think it's ok to check the duplicator before building the event?

Reply:
Yeah, I can sacrifice ergonomics for performance here